From 333b2f4660482c03fb45fc6f900e9cf8c14feff4 Mon Sep 17 00:00:00 2001
From: Michal Srb <michalsrb@gmail.com>
Date: Tue, 21 Feb 2017 22:57:32 +0200
Subject: [PATCH 04/13] Add FcStrHashIgnoreBlanksAndCase.

Creates hash of string ignoring blanks and case. If two strings are compared equal using FcStrCmpIgnoreBlanksAndCase function, they will have the same hash using FcStrHashIgnoreBlanksAndCase.
---
 src/fcint.h |  3 +++
 src/fcstr.c | 13 +++++++++++++
 2 files changed, 16 insertions(+)

diff --git a/src/fcint.h b/src/fcint.h
index a899480..0799961 100644
--- a/src/fcint.h
+++ b/src/fcint.h
@@ -1202,6 +1202,9 @@ FcStrLastSlash (const FcChar8  *path);
 FcPrivate FcChar32
 FcStrHashIgnoreCase (const FcChar8 *s);
 
+FcPrivate FcChar32
+FcStrHashIgnoreBlanksAndCase (const FcChar8 *s);
+
 FcPrivate FcChar8 *
 FcStrCanonFilename (const FcChar8 *s);
 
diff --git a/src/fcstr.c b/src/fcstr.c
index b65492d..4d7fc8a 100644
--- a/src/fcstr.c
+++ b/src/fcstr.c
@@ -256,6 +256,19 @@ FcStrHashIgnoreCase (const FcChar8 *s)
     return h;
 }
 
+FcChar32
+FcStrHashIgnoreBlanksAndCase (const FcChar8 *s)
+{
+    FcChar32	    h = 0;
+    FcCaseWalker    w;
+    FcChar8	    c;
+
+    FcStrCaseWalkerInit (s, &w);
+    while ((c = FcStrCaseWalkerNext (&w, " ")))
+	h = ((h << 3) ^ (h >> 3)) ^ c;
+    return h;
+}
+
 /*
  * Is the head of s1 equal to s2?
  */
-- 
2.15.1

